### ISSUE TITLE
App crash with "out of memory"

### ISSUE THREAD

**Issue by thot1** [deckerst/aves#688]
**Describe the bug**
The App crashes seconds after start

**To Reproduce**
Start the app --> wait some seconds --> crash

**Expected behavior**

**Screenshots**

**System information and logs:**
```
type: crash
osVersion: google/oriole/oriole:13/TQ3A.230705.001/2023071100:user/release-keys
package: deckers.thibault.aves.libre:10002
process: deckers.thibault.aves.libre
processUptime: 4809 + 206 ms

java.lang.OutOfMemoryError: Failed to allocate a 55380616 byte allocation with 6163528 free bytes and 6019KB until OOM, target footprint 268435456, growth limit 268435456
	at java.util.Arrays.copyOf(Arrays.java:3645)
	at java.util.Arrays.copyOf(Arrays.java:3614)
	at java.util.ArrayList.grow(ArrayList.java:275)
	at java.util.ArrayList.ensureExplicitCapacity(ArrayList.java:249)
	at java.util.ArrayList.ensureCapacityInternal(ArrayList.java:241)
	at java.util.ArrayList.add(ArrayList.java:467)
	at t3.c.a(Unknown Source:4)
	at t3.e.a(Unknown Source:108)
	at t3.a.a(Unknown Source:54)
	at s6.a.s(Unknown Source:60)
	at s6.a.r(Unknown Source:64)
	at t6.d.g(Unknown Source:36)
	at t6.d.i(Unknown Source:38)
	at u6.p.U(Unknown Source:496)
	at u6.p.V(Unknown Source:35)
	at u6.p.T(Unknown Source:141)
	at p6.w.i(Unknown Source:22)
	at p6.w.e(Unknown Source:0)
	at p6.w$c.invokeSuspend(Unknown Source:12)
	at e8.a.resumeWith(Unknown Source:11)
	at u8.r0.run(Unknown Source:93)
	at z8.o$a.run(Unknown Source:3)
	at b9.k.run(Unknown Source:2)
	at b9.a.x(Unknown Source:0)
	at b9.a$c.d(Unknown Source:14)
	at b9.a$c.p(Unknown Source:28)
	at b9.a$c.run(Unknown Source:0)
	Suppressed: z8.i: [f2{Cancelling}@dd1860, Dispatchers.IO]

```

**Additional context**

**Comment by deckerst** [comment]
Thanks for the report.

Please start the app in `safe mode` (it's one of the app shortcuts), then go to `About` > `Bug Report` and follow the instructions to get your system info and logs, so you can attach them here.

**Comment by thot1** [comment]
Unfortunately, the app crashes even in safe mode. Unfortunately, it is not possible to save the logs. Only the system information can be copied quickly.

Package: deckers.thibault.aves.libre
Installer: com.android.packageinstaller
Aves version: 1.8.9-libre, build 10002
Flutter: stable 3.10.3
Android version: 13, API 33
Android build: TQ3A.230705.001.2023071100
Device: Google Pixel 6
Geocoder: not available
Mobile services: not available
System locales: de_DE
Storage volumes: /storage/emulated/0/
Storage grants: 
Error reporting: false

**Comment by deckerst** [comment]
Damned, i had designed the safe mode specifically for cases like yours, when the analysis makes the app crash before the user can do anything. In theory the safe mode disables the analysis :/

An alternative is for you to install the Play version, with automatic error reporting enabled, so that i can see the crash logs reported there. Knowing your device and environment details i should be able to find the report among others.

**Comment by thot1** [comment]
I am not sure if this will work. I do not have Google Services installed on my smartphone.... Is the app to be installed the second available in the F-Droid Store?

**Comment by deckerst** [comment]
> I am not sure if this will work. I do not have Google Services installed on my smartphone.... Is the app to be installed the second available in the F-Droid Store?

No, that one is probably izzy's version, which doesn't include error reporting either.

**Comment by thot1** [comment]
Ok, I fixed the problem with the app crashing. Maybe there is a problem with the memory access in the App. But as I descripted, these problems startet after GrapheneOS system update. Since this was not the only problem, I decide to take a factory reset and import a data backup. After that, the app no longer crashes.
So, I think we can solve this issue! Many thanks for your fast support and for this great app. There is definitly a donatation from me!!!
